runtime/test/manifest-test.js

```diff
@@ -21,27 +21,41 @@ async function assertRecipeParses(input, result) {
   assert.deepEqual((await Manifest.parse(input)).recipes[0].toString(), target);
 }
 
+async function assertManifestParses(manifestStr, options, handler) {
+  let manifest = await Manifest.parse(manifestStr, options);
+  if (handler) {
+    handler(manifest);
+  }
+
+  let deserializedManifest = await Manifest.parse(manifest.toString(), options);
+  if (handler) {
+    handler(deserializedManifest);
+  }
+}
+
 describe('manifest', function() {
   it('can parse a manifest containing a recipe', async () => {
-    let manifest = await Manifest.parse(`
-      particle SomeParticle
+    await assertManifestParses(`
```

Comment:
Maybe this should return the parsed manifest? to avoid having to both pass a continuation and await it.

Reply:
I'm not sure I understand what you suggest - 
the reason i refactored this into a helper method was to run all the asserts twice - on the parsed manifest, and then on the manifest parsed from serialized manifest.
the reason i await the helper method is for the test to fail, if any of the asserts fails.

alternative way to do the same would be

let verify = (manifest) => { /* run all the relevant asserts */ }
let manifest = await Manifest.parse(`...`, {/* options */});
verify(manifest);
let deserializedManifest = await Manifest.parse(manifest.toString(), {/* options */});
verify(manifest);

this is a bit more verbose, but probably more readable. 

Let me know what you prefer. Or did you mean smth totally different?

Thanks.

Reply:
Ah, sorry I didn't look that carefully. But i wonder if it would be OK to just run the asserts on the final manifest.

```
let manifest = await assertManifestParses(...);
assert(manifest...);
```

I find test utils that call callbacks with asserts a bit tricky to debug when they fail. But maybe it's OK in this case.